src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -271,27 +283,71 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 				continue;
 			}
 
-			if (contain_mutable_functions((Node *) targetEntry->expr))
+			if (commandType == CMD_UPDATE &&
+				contain_volatile_functions((Node *) targetEntry->expr))
 			{
-				hasNonConstTargetEntryExprs = true;
+				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+								errmsg("functions used in UPDATE queries on distributed "
```

Comment:
Well, the same's true for INSERTs with more than one row, or inserts from SELECT. Probably worth adding a big comment about, given that both are likely to be supported in the not too far away future.